akka-http/src/main/scala/akka/http/scaladsl/server/HttpApp.scala

```diff
@@ -132,10 +132,15 @@ abstract class HttpApp extends Directives {
    * Hook that lets the user specify the future that will signal the shutdown of the server whenever completed.
    */
   protected def waitForShutdownSignal(system: ActorSystem)(implicit ec: ExecutionContext): Future[Done] = {
+    val promise = Promise[Done]()
+    sys.addShutdownHook {
+      promise.success(Done)
+    }
     Future {
-      StdIn.readLine("Press RETURN to stop...\n")
-      Done
+      if (StdIn.readLine("Press RETURN to stop...\n") != null)
```

Comment:
Do you really need the if in this case? `StdIn.readline` is blocking anyway and no matter what it returns we want the promise to complete.